contracts/bridge/MiniChefV21.sol

```diff
@@ -0,0 +1,444 @@
+        .sub(user.rewardDebt)
+        .toUInt256();
+
+        // Effects
+        user.rewardDebt = accumulatedSynapse;
+
+        if (_pendingSynapse != 0) {
+            totalPendingSynapse = totalPendingSynapse.sub(_pendingSynapse);
+            // Interactions start here
+            SYNAPSE.safeTransfer(to, _pendingSynapse);
+        }
+
+        IRewarder _rewarder = rewarder[pid];
+        if (address(_rewarder) != address(0)) {
+            // We need bonusFarm to recognize that bonusReward needs to be sent.
+            // Rewards are only sent when _pendingSynapse > 0.
+            // So we set _pendingSynapse to 1 wei, if there are no pending rewards.
+            // Now in case the emissions for main pool are shut off before
+            // the bonus period ends, anyone will be able to claim their bonus rewards.
+            _rewarder.onSynapseReward(
+                pid,
+                msg.sender,
+                to,
+                _pendingSynapse != 0 ? _pendingSynapse : 1,
+                user.amount // amount isn't modified in harvest()
+            );
+        }
+
+        emit Harvest(msg.sender, pid, _pendingSynapse);
+    }
+
+    /// @notice Withdraw LP tokens from MCV2 and harvest proceeds for transaction sender to `to`.
+    /// @param pid The index of the pool. See `poolInfo`.
+    /// @param amount LP token amount to withdraw.
+    /// @param to Receiver of the LP tokens and SYNAPSE rewards.
+    function withdrawAndHarvest(
+        uint256 pid,
+        uint256 amount,
+        address to
+    ) external {
+        PoolInfo memory pool = updatePool(pid);
+        UserInfo storage user = userInfo[pid][msg.sender];
+        uint256 oldAmount = user.amount;
+        int256 accumulatedSynapse = int256(
+            user.amount.mul(pool.accSynapsePerShare) / ACC_SYNAPSE_PRECISION
+        );
+        uint256 _pendingSynapse = accumulatedSynapse
+        .sub(user.rewardDebt)
+        .toUInt256();
+
+        // Effects
+        user.rewardDebt = accumulatedSynapse.sub(
+            int256(amount.mul(pool.accSynapsePerShare) / ACC_SYNAPSE_PRECISION)
+        );
+        user.amount = user.amount.sub(amount);
+
+        if (_pendingSynapse != 0) {
+            totalPendingSynapse = totalPendingSynapse.sub(_pendingSynapse);
+            // Interactions start here
+            SYNAPSE.safeTransfer(to, _pendingSynapse);
+        }
+
+        IRewarder _rewarder = rewarder[pid];
+        if (address(_rewarder) != address(0)) {
+            // see harvest() for _pendingSynapse explanation
+            _rewarder.onSynapseReward(
+                pid,
+                msg.sender,
+                to,
+                _pendingSynapse != 0 ? _pendingSynapse : 1,
+                oldAmount
+            );
+        }
+
+        lpToken[pid].safeTransfer(to, amount);
+
+        emit Withdraw(msg.sender, pid, amount, to);
+        emit Harvest(msg.sender, pid, _pendingSynapse);
```

Comment:
(Not actually a review comment, feel free to resolve without responding)

A single event emit might make more sense here, something like `event(WithdrawAndHarvest(sender, pid, amount, to, _pendingSynapse_`

Reply:
Generally you want to keep events standardized since they'll primarily be used by off-chain indexers, so the less events / the more standard, the better.

Reply:
Agree that it's better to have `Withdraw` and `Harvest` events w/o complex `WithdrawAndHarvest`. Let's not make it any harder than it should be.